Source/cryptalgo/SecureSocketPort.cpp

```diff
@@ -130,43 +130,60 @@ uint32_t SecureSocketPort::Handler::Initialize() {
     uint32_t success = Core::ERROR_NONE;
 
     if (IsOpen() == true) {
-        _context = SSL_CTX_new(TLS_server_method());
         _handShaking = ACCEPTING;
-    }
-    else {
-        _context = SSL_CTX_new(TLS_method());
+    } else {
         _handShaking = CONNECTING;
     }
 
-    _ssl = SSL_new(static_cast<SSL_CTX*>(_context));
-    SSL_set_fd(static_cast<SSL*>(_ssl), static_cast<Core::IResource&>(*this).Descriptor());
-    SSL_CTX_set_options(static_cast<SSL_CTX*>(_context), SSL_OP_ALL | SSL_OP_NO_SSLv2);
+    // Client and server use
+    _context = SSL_CTX_new(TLS_method());
 
-    // Trust the same certificates as any other application
-    if (SSL_CTX_set_default_verify_paths(static_cast<SSL_CTX*>(_context)) == 1) {
-        success = Core::SocketPort::Initialize();
+    ASSERT(_context != nullptr);
 
-        if (success == Core::ERROR_NONE) {
-            SSL_set_tlsext_host_name(static_cast<SSL*>(_ssl), RemoteNode().HostName().c_str());
-        }
-    }
-    else {
-        TRACE_L1("OpenSSL failed to load certificate store");
+    constexpr uint64_t options = SSL_OP_ALL | SSL_OP_NO_SSLv2;
+
+    VARIABLE_IS_NOT_USED uint64_t bitmask = SSL_CTX_set_options(static_cast<SSL_CTX*>(_context), options);
+
+    ASSERT((bitmask & options) == options);
+
+    if ( ( (    (_handShaking == ACCEPTING)
+              // Load server certifiate and private key
+             && (SSL_CTX_load_verify_locations(static_cast<SSL_CTX*>(_context), "RootCA.pem", nullptr) == 1)
```

Comment:
The missing code in case we are a server :-) I would assume that these three files, "RootCA.pem", "localhost.pem" and "localhost.key" can be made a parameter and I guess/hope we can make the loading of these certificates part of the constructor (of the server) to validate their "loadability" and if not loadable, just stop the server from opening up..